tests/sentry/profiles/test_task.py

```diff
@@ -51,3 +105,194 @@ def test_normalize_android_profile(self):
+            },
+        )
+
+        self.login_as(user=self.owner)
+
+        out = BytesIO()
+        with ZipFile(out, "w") as f:
+            f.writestr(f"proguard/{PROGUARD_INLINE_UUID}.txt", PROGUARD_INLINE_SOURCE)
+
+        response = self.client.post(
+            url,
+            {
+                "file": SimpleUploadedFile(
+                    "symbols.zip", out.getvalue(), content_type="application/zip"
+                )
+            },
+            format="multipart",
+        )
+        assert response.status_code == 201, response.content
+        assert len(response.data) == 1
+
+        profile = dict(self.android_profile)
+        profile.update(
+            {
+                "build_id": PROGUARD_INLINE_UUID,
+                "project_id": self.project.id,
+                "profile": {
+                    "methods": [
+                        {
+                            "name": "onClick",
+                            "abs_path": None,
+                            "class_name": "e.a.c.a",
+                            "source_file": None,
+                            "source_line": 2,
+                        },
+                        {
+                            "name": "t",
+                            "abs_path": None,
+                            "class_name": "io.sentry.sample.MainActivity",
+                            "source_file": "MainActivity.java",
+                            "source_line": 1,
+                        },
+                    ],
+                },
+            }
+        )
+
+        profile = _deobfuscate(profile)
+        frames = profile["profile"]["methods"]
+
+        assert sum(len(f.get("inline_frames", [{}])) for f in frames) == 4
+
+        assert frames[0]["name"] == "onClick"
+        assert frames[0]["class_name"] == "io.sentry.sample.-$$Lambda$r3Avcbztes2hicEObh02jjhQqd4"
+
+        assert frames[1]["inline_frames"][0]["source_file"] == "MainActivity.java"
+        assert frames[1]["inline_frames"][0]["class_name"] == "io.sentry.sample.MainActivity"
+        assert frames[1]["inline_frames"][0]["name"] == "bar"
+        assert frames[1]["inline_frames"][0]["source_line"] == 54
+        assert frames[1]["inline_frames"][1]["name"] == "foo"
+        assert frames[1]["inline_frames"][1]["source_line"] == 44
+        assert frames[1]["inline_frames"][2]["name"] == "onClickHandler"
+        assert frames[1]["inline_frames"][2]["source_line"] == 40
+        assert frames[1]["inline_frames"][2]["source_file"] == "MainActivity.java"
+        assert frames[1]["inline_frames"][2]["class_name"] == "io.sentry.sample.MainActivity"
+
+    def test_error_on_resolving(self):
+        url = reverse(
+            "sentry-api-0-dsym-files",
+            kwargs={
+                "organization_slug": self.project.organization.slug,
+                "project_slug": self.project.slug,
+            },
+        )
+
+        self.login_as(user=self.owner)
```

Comment:
you can move some of this into the setup to avoid having to repeat it in each test